src/DataCollector/GuzzleCollector.php

```diff
@@ -64,6 +64,10 @@ public function collect(Request $request, Response $response, \Exception $except
         /** @var RequestInterface $request */
         foreach ($this->history as $request) {
             $transaction = $this->history[$request];
+            if (!isset($transaction['response'])) {
+                continue;
```

Comment:
This won't actually fix things. You may run into cases where you have no response, but still an error.